### PR TITLE
feat: Implement Tabs component with storybook #14

### PR DESCRIPTION
**Pull request by dgd03146**
## Motivation 🤔

- Implementing a Tabs component that ensures ease of access and complies with accessibility standards.
- Utilizing Storybook to demonstrate the functionality and accessibility of the Tabs component.
- Reference Design: [UI Design for Tabs](https://ui.shadcn.com/docs/components/tabs)

<br/>

## Key Changes 🔑

- Created a `Tabs` component structured with accessibility in mind, including appropriate ARIA roles and properties.
- Developed `TabsList`, `TabsTrigger`, and `TabsContent` subcomponents to manage tabs' state and interactions.
- Added Storybook stories to illustrate different states and props for the Tabs component.
- Integrated Storybook's accessibility add-ons to ensure the component meets accessibility standards (WCAG).

<br/>

## To Reviewers 🙏🏻

@KIMSEUNGGYU 
- Please review the structure and logic of the new Tabs components, especially the use of context for state management.
- Evaluate the accessibility features, including keyboard navigation and ARIA attributes.
- Assess the Storybook integration, especially the use of add-ons for accessibility checks.
- Suggestions for improving code readability and efficiency are highly welcome.

<br/>

### CODE REVIEW

File: packages/design-system/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import type { OmitAtomProps } from '../../types/atoms';
+import { Box } from '../Box';
+import { TabsContext } from './TabsContext';
+
+export interface TabsProps
+	extends Omit<HTMLAttributes<HTMLDivElement>, 'defaultValue'>,
+		OmitAtomProps {
+	defaultValue?: string | number;
+}
+
```
Comment (KIMSEUNGGYU) [inline packages/design-system/components/Tabs/Tabs.tsx]: HTML Div Element 안에 defaultValue 라는 속성이 있나요??

File: packages/design-system/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export const Tabs = forwardRef<HTMLDivElement, TabsProps>(
+	({ children, defaultValue, ...restProps }, ref?) => {
+		const [currentTab, setCurrentTab] = useState(defaultValue);
+		const contextValue = useMemo(
+			() => ({
+				currentTab,
+				setCurrentTab,
+			}),
+			[currentTab],
+		);
```
Comment (KIMSEUNGGYU) [inline packages/design-system/components/Tabs/Tabs.tsx:22]: 오 Memo 로 성능까지 신경쓰셨군요 👍🏻👍🏻👍🏻👍🏻

File: packages/design-system/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { TabsContext } from './TabsContext';
+
+export interface TabsProps
+	extends Omit<HTMLAttributes<HTMLDivElement>, 'defaultValue'>,
+		OmitAtomProps {
+	defaultValue?: string | number;
+}
+
+export const Tabs = forwardRef<HTMLDivElement, TabsProps>(
+	({ children, defaultValue, ...restProps }, ref?) => {
```
Comment (KIMSEUNGGYU) [inline packages/design-system/components/Tabs/Tabs.tsx]: `ref?` 으로 처리한 이유가 무엇인가요??

**Review by KIMSEUNGGYU — APPROVED** [review]
좋네요!

File: packages/design-system/components/Tabs/Tabs.tsx (hunk 3 above)
Comment (dgd03146) [reply packages/design-system/components/Tabs/Tabs.tsx]: ref를 받을수도 있고 안 받을수도 있어서 그렇게 처리하였는데 승규님 말씀듣고 찾아보니
image
이미 ref는 optional이군요 ㅎㅎ 짚어주셔서 감사합니다.

File: packages/design-system/components/Tabs/Tabs.tsx (hunk 1 above)
Comment (dgd03146) [reply packages/design-system/components/Tabs/Tabs.tsx]: 그때 타입 오류가 났었던 것 같은데 다시 찾아보니까 없네요. 수정해놓겠습니다.

**Pull request merged**